src/pages/home.jsx

```diff
@@ -1,46 +1,70 @@
-import React, { useContext, useEffect, useState } from 'react';
-import { UserContext } from '../context/userContext';
+import React, {useEffect, useState } from 'react';
+
 import PostSection from '../components/PostSection';
 import { PICKLY } from '../apis/clients/pickly';
 import PostLoader from '../components/LoadingComponents/PostLoader';
 
 export const Home = () => {
-  const { token } = useContext(UserContext);
-  const [data, setData] = useState(null);
+  const [posts, setPosts] = useState(null);
   const [loading, setLoading] = useState(false);
   const [error, setError] = useState(false);
+  const postsNum = 10;
+  let pageNum = 1;
+
   // This useEffect() for fetching data when the route load
+
   useEffect(() => {
     setLoading(true);
-    PICKLY.getAllPosts()
+    PICKLY.getAllPosts(pageNum, postsNum)
       .then(({ data }) => {
-        setData(data.data);
+        setPosts(data.data);
         setLoading(false);
       })
       .catch(err => {
-        setLoading(false);
         setError(true);
+        setLoading(false);
       });
-  }, [token]);
+
+    const fetchNewPost = () => {
+      setLoading(true);
+      pageNum++;
+
+      PICKLY.getAllPosts(pageNum, postsNum)
+        .then(({ data }) => {
+          setPosts(prevPosts => {
+            return prevPosts.concat(data.data);
+          });
+          setLoading(false);
+        })
+        .catch(err => {
+          setLoading(false);
+          setError(true);
+        });
+    };
+    document.addEventListener('scroll', () => {
```

Comment:
please ensure that whenever you add an event listener, you remove it when the component got unmounted/disappeared